From 2f8a2e58d72518bdf97a249d5387caadc09264d1 Mon Sep 17 00:00:00 2001
From: Virtroid build <build@virtroid.invalid>
Date: Sat, 31 Oct 2026 13:37:36 +0000
Subject: [PATCH] Allow the x264 screen encoder in the swcodec APEX

libcodec2_soft_x264enc (frameworks/av, c2.virtroid.x264.encoder) and the
x264 it links, prebuilt_libx264_virtroid (device/virtroid/x264), are new
dependencies of com.android.media.swcodec; the APEX dependency check
fails the build until they are listed here. Both as the generated list
has them (update-apex-allowed-deps.sh).

Change-Id: I29b28151f1a13ed724d378cc1eeedc9dbd5fa591
---
 build/allowed_deps.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/allowed_deps.txt b/build/allowed_deps.txt
index bc841d7..9aa9f1e 100644
--- a/build/allowed_deps.txt
+++ b/build/allowed_deps.txt
@@ -786,6 +786,7 @@ libcodec2_soft_vp8dec(minSdkVersion:29)
 libcodec2_soft_vp8enc(minSdkVersion:29)
 libcodec2_soft_vp9dec(minSdkVersion:29)
 libcodec2_soft_vp9enc(minSdkVersion:29)
+libcodec2_soft_x264enc(minSdkVersion:29)
 libcodec2_vndk(minSdkVersion:29)
 libcom_android_networkstack_tethering_util_jni(minSdkVersion:30)
 libcombine(minSdkVersion:29)
@@ -1600,6 +1601,7 @@ prebuilt_libnativehelper_compat_libc++(minSdkVersion:(no version))
 prebuilt_libnativehelper_header_only(minSdkVersion:(no version))
 prebuilt_libperfetto_client_experimental(minSdkVersion:(no version))
 prebuilt_libunwind(minSdkVersion:(no version))
+prebuilt_libx264_virtroid(minSdkVersion:29)
 prebuilt_ndk_libc++_static(minSdkVersion:(no version))
 prebuilt_ndk_libc++abi(minSdkVersion:(no version))
 prebuilt_perfetto_trace_protos(minSdkVersion:(no version))
-- 
2.43.0

